cpp/examples/advanced/cugraph_operations/graph_operations.cu

```diff
@@ -0,0 +1,253 @@
+ * display vertex and edge partitions.
+ */
+
+template <typename vertex_t, typename edge_t, typename weight_t, bool multi_gpu>
+void perform_example_graph_operations(raft::handle_t const& handle,
+                                      std::string const& csv_graph_file_path,
+                                      const bool weighted = false)
+{
+  auto const comm_rank = handle.get_comms().get_rank();
+  auto const comm_size = handle.get_comms().get_size();
+
+  std::cout << "Rank_" << comm_rank << ", reading graph from " << csv_graph_file_path << std::endl;
+
+  bool renumber = true;  // must be true for distributed graph.
+
+  // Read a graph (along with edge properties e.g. edge weights, if provided) from
+  // the input csv file
+
+  auto [graph, edge_weights, renumber_map] =
+    cugraph::test::read_graph_from_csv_file<vertex_t, edge_t, weight_t, false, multi_gpu>(
+      handle, csv_graph_file_path, weighted, renumber);
+
+  // Non-owning view of the graph object
+  auto graph_view = graph.view();
+  // Number of vertices mapped to this process, ie the size of
+  // the vertex partition assigned to this process
+
+  vertex_t size_of_the_vertex_partition_assigned_to_this_process =
+    graph_view.local_vertex_partition_range_size();
+
+  // Non-owning of the edge edge_weights object
+  auto edge_weight_view = edge_weights ? std::make_optional((*edge_weights).view()) : std::nullopt;
+
+  using graph_view_t = cugraph::graph_view_t<vertex_t, edge_t, false, multi_gpu>;
+
+  //
+  // As an example operation, compute the weighted average of the properties of
+  // neighboring vertices, weighted by the edge weights, if the input graph is weighted;
+  // Otherwise, compute the simple average.
+  //
+  if (weighted) {
+    using result_t      = weight_t;
+    auto vertex_weights = compute_out_weight_sums(handle, graph_view, *edge_weight_view);
+
+    cugraph::edge_src_property_t<graph_view_t, result_t> src_vertex_weights_cache(handle,
+                                                                                  graph_view);
+
+    cugraph::edge_dst_property_t<graph_view_t, result_t> dst_vertex_weights_cache(handle,
+                                                                                  graph_view);
+
+    update_edge_src_property(handle, graph_view, vertex_weights.begin(), src_vertex_weights_cache);
+
+    update_edge_dst_property(handle, graph_view, vertex_weights.begin(), dst_vertex_weights_cache);
+
+    rmm::device_uvector<result_t> outputs(size_of_the_vertex_partition_assigned_to_this_process,
+                                          handle.get_stream());
+
+    per_v_transform_reduce_incoming_e(
+      handle,
+      graph_view,
+      src_vertex_weights_cache.view(),
+      dst_vertex_weights_cache.view(),
+      (*edge_weight_view),
+      [] __device__(auto src, auto dst, auto src_prop, auto dst_prop, auto edge_prop) {
+        printf("\nsrc ---> %d dst = %d :  src_prop = %f dst_prop = %f edge_prop = %f\n",
```

Comment:
What is "\nsrc ---> %d dst = %d"? Did you intend "src = %d ---> dst = %d"? And why "\n" in both the beginning and the end?

Reply:
Same comments for the else (no edge weight) path.

Reply:
Changed.